tools/codemod/src/utils/typeGuards.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type {
+  ArrayExpression, ASTNode, CallExpression, Identifier, ImportSpecifier, JSCodeshift, JSXAttribute, JSXElement, JSXEmptyExpression,
+  JSXExpressionContainer, JSXIdentifier, Literal, MemberExpression, ObjectExpression,
+} from "jscodeshift";
+
+type Source = Parameters<JSCodeshift>[0];
+
+export function isJSXIdentifierType(j: JSCodeshift, source: Source): source is JSXIdentifier {
```

Comment:
Only type guards that take `ASTNode` as source are correct, @AntanasBuk is working on refactoring this.